libcni/api.go

```diff
@@ -57,27 +63,63 @@ type CNIConfig struct {
 // CNIConfig implements the CNI interface
 var _ CNI = &CNIConfig{}
 
-func buildOneConfig(list *NetworkConfigList, orig *NetworkConfig, prevResult types.Result) (*NetworkConfig, error) {
+func buildOneConfig(list *NetworkConfigList, orig *NetworkConfig, prevResult types.Result, rt *RuntimeConf) (*NetworkConfig, error) {
 	var err error
 
+	inject := map[string]interface{}{
+		"name":       list.Name,
+		"cniVersion": list.CNIVersion,
+	}
+	// Add previous plugin result
+	if prevResult != nil {
+		inject["prevResult"] = prevResult
+	}
+
 	// Ensure every config uses the same name and version
-	orig, err = InjectConf(orig, "name", list.Name)
+	orig, err = InjectConf(orig, inject)
 	if err != nil {
 		return nil, err
 	}
-	orig, err = InjectConf(orig, "cniVersion", list.CNIVersion)
-	if err != nil {
-		return nil, err
+
+	return injectRuntimeConfig(orig, rt)
+}
+
+func injectRuntimeConfig(orig *NetworkConfig, rt *RuntimeConf) (*NetworkConfig, error) {
```

Comment:
Can you add a quick comment that explains what this function does? It's just complicated (and "magic") enough that a quick overview would be helpful.

I suspect this is the logic that non-go runtimes might have the most trouble with.